llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8336,6 +8333,14 @@ void LoopVectorizationPlanner::buildVPlansWithVPRecipes(ElementCount MinVF,
       if (CM.foldTailWithEVL() && !HasScalarVF)
         VPlanTransforms::runPass(VPlanTransforms::addExplicitVectorLength,
                                  *Plan, CM.getMaxSafeElements());
+
+      if (auto P = VPlanTransforms::narrowInterleaveGroups(
+              *Plan,
+              TTI.getRegisterBitWidth(
+                  TargetTransformInfo::RGK_FixedWidthVector),
```

Comment:
How does this work if `Plan` is for a scalable vector? I thought with https://github.com/llvm/llvm-project/pull/154842 we now supported narrow interleaving groups for scalable VFs?

Reply:
At the moment, I think for all cases in practice, known min value for the fixed and scalable vectors is the same (128 on AArch64). 

But they could be different in theory, so I moved retrieving the bitwidth into `narrowInterleaveGroups`,  depending on whether the VF we are trying to optimize is scalable or fixed.